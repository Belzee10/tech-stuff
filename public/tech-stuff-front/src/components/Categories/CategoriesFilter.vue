<template>
  <div class="categories-filter">
    <v-checkbox
      v-for="item in categories"
      :key="item.id"
      v-model="value"
      :label="item.name"
    ></v-checkbox>
  </div>
</template>

<script>
export default {
  name: 'CategoriesFilter',
  props: {
    categories: {
      type: Array,
      default: () => []
    },
    value: {
      type: Boolean,
      default: false
    }
  }
};
</script>

<style lang="scss" scoped></style>
