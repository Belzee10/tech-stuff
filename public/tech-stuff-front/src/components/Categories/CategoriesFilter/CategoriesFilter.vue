<template>
  <div class="categories-filter pa-3">
    <v-radio-group :value="value" @change="value => $emit('check', value)">
      <v-radio
        v-for="item in categories"
        :key="item.id"
        color="primary"
        class="mb-2 category-check"
        :label="item.name"
        :value="item.id"
      ></v-radio>
    </v-radio-group>
  </div>
</template>

<script>
export default {
  name: 'CategoriesFilter',
  props: {
    categories: {
      type: Array,
      default: () => []
    },
    value: {
      type: [Number, String],
      default: 0
    }
  }
};
</script>

<style lang="scss"></style>
