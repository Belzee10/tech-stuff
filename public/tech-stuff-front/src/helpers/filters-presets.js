export default {
  categories: ['id', 'name'],
  products: [
    'id',
    'name',
    'price',
    'image',
    'brand',
    'description',
    'category_id'
  ],
  orders: [],
  user: []
};
