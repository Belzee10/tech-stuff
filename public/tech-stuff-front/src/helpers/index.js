import preset from './filters-presets.js';
/**
 * generate an array
 * @param {Number} count
 * @param {Object} item
 */
export const generateArray = (count, item = {}) => {
  const array = new Array(count);
  return array.fill().map((_, index) => {
    return typeof item === 'function'
      ? item(index)
      : {
          ...item,
          id: index
        };
  });
};

/**
 * generate random integer
 * @param {Number} from
 * @param {Number} to
 */
export const generateNumber = (from, to) =>
  Math.floor(Math.random() * (to - from + 1) + from);

/**
 * filter the expected object properties
 * @param {Object} obj
 * @param {*} exp
 */
export const filterObject = (obj, exp) => {
  const result = {};
  Object.keys(obj).forEach(key => {
    let expValue;
    if (Array.isArray(exp)) expValue = exp;
    else if (typeof exp === 'string') expValue = preset[exp];
    if (expValue.includes(key)) result[key] = obj[key];
  });
  return result;
};
