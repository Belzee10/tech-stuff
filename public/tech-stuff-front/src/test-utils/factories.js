import { mount, createLocalVue } from '@vue/test-utils';
import mergeWith from 'lodash.mergewith';
import Vuetify from 'vuetify';
import Vuex from 'vuex';

const localVue = createLocalVue();
localVue.use(Vuex);

let vuetify;

beforeEach(() => {
  vuetify = new Vuetify();
});

const customizer = (_, srcValue) => {
  if (Array.isArray(srcValue)) {
    return srcValue;
  }
  if (srcValue instanceof Object && Object.keys(srcValue).length === 0) {
    return srcValue;
  }
};

/**
 * create component wrapper
 * @param {*} Component
 * @param {Object} overrides
 */
const createWrapper = (Component, overrides) => {
  const defaultMountingOptions = {
    localVue,
    vuetify
  };
  return mount(
    Component,
    mergeWith(defaultMountingOptions, overrides, customizer)
  );
};

/**
 * create Vuex store
 * @param {Object} overrides
 */
const createStore = overrides => {
  const defaultStoreConfig = {};
  return new Vuex.Store(mergeWith(defaultStoreConfig, overrides, customizer));
};

export { createWrapper, createStore };
