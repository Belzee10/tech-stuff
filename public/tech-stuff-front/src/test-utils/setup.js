import Vue from 'vue';
import Vuetify from 'vuetify';

Vue.config.productionTip = false;

Vue.use(Vuetify);

/**
 * silent "console.error"
 */
global.console = {
  error: jest.fn(),
  log: console.log,
  warn: console.warn,
  info: console.info,
  debug: console.debug
};
