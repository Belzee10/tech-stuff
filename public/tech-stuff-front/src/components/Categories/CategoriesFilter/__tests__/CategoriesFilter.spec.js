import { createWrapper } from '@/test-utils/factories';
import CategoriesFilter from '../CategoriesFilter.vue';
import { generateArray } from '@/helpers';

describe('CategoriesFilter.vue', () => {
  test('should render all "categories"', () => {
    const props = {
      categories: generateArray(1, {
        name: '',
        value: ''
      })
    };
    const wrapper = createWrapper(CategoriesFilter, {
      propsData: props
    });
    const categories = wrapper.findAll('.v-input');
    expect(categories).toHaveLength(props.categories.length);
  });

  test('should set checked the "value" element', () => {
    const props = {
      categories: generateArray(2, i => ({
        id: i,
        name: `Cat ${i}`
      }))
    };
    const wrapper = createWrapper(CategoriesFilter, {
      propsData: {
        ...props,
        value: 1
      }
    });
    const firstCategory = wrapper.find('.v-radio:last-child input');
    expect(firstCategory.attributes('value')).toBe('1');
  });

  test('should render correctly', () => {
    const props = {
      categories: generateArray(1, {
        name: ''
      })
    };
    const wrapper = createWrapper(CategoriesFilter, {
      propsData: props
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
