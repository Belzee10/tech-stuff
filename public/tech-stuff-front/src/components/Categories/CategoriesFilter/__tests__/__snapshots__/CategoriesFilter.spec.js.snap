// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesFilter.vue should render correctly 1`] = `
<div
  class="categories-filter pa-3"
>
  <div
    class="v-input theme--light v-input--selection-controls v-input--radio-group v-input--radio-group--column"
  >
    <div
      class="v-input__control"
    >
      <div
        class="v-input__slot"
        style="height: auto;"
      >
        <div
          aria-labelledby="input-21"
          class="v-input--radio-group__input"
          role="radiogroup"
        >
          <div
            class="v-radio mb-2 category-check theme--light v-item--active"
          >
            <div
              class="v-input--selection-controls__input"
            >
              <input
                aria-checked="true"
                id="input-22"
                name="radio-21"
                role="radio"
                type="radio"
                value="0"
              />
              <div
                class="v-input--selection-controls__ripple primary--text"
              />
              <i
                aria-hidden="true"
                class="v-icon notranslate mdi mdi-radiobox-marked theme--light primary--text"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="v-messages theme--light"
      >
        <span
          class="v-messages__wrapper"
          name="message-transition"
          tag="div"
        />
      </div>
    </div>
  </div>
</div>
`;
